tests/test_functions.py

```diff
@@ -73,21 +73,31 @@ def compare_request(requestcontent, request, requesttype):
         
         changes = difflib.unified_diff(olddata, newdata)
 
-        # temp variable to detect if we need to print the beginning of the error
-        firstchangep = True
+        
 
         # change list holds the strings to print in an error message
         changelist = [requesttype, " ", request, " did not match previous results. If you are adding changes to SynBioHub that change this page, please check that the page is correct and update the file using the command line argument --resetgetrequests [requests] and --resetpostrequests [requests].\nThe following is a diff of the new files compared to the old.\n"]
+
+        # temp variable to detect if we need to print the beginning of the error
+        numofchanges = 0
         
         for c in changes:
-            if firstchangep:
-                firstchangep = False
+            numofchanges += 1
 
             # print the diff
             changelist.append(c)
             changelist.append("\n")
 
-        if not firstchangep:
+        # if it was only a one line change, there are 11 lines in the diff
```

Comment:
Can you add an HTML id to the div containing the whole version string, then use BeautifulSoup to empty out the string? That seems cleaner and will be more similar to the approach you'd use to ignore other parts of the page. Writing a regex for each thing to ignore will get unwieldy.

Reply:
That seems like a much better solution- I'll work on that.

Reply:
It seems that an id in html is unique- I will use a class called testignore instead because then multiple items can be of the class. Is that a good way to do it?